### ISSUE TITLE
With the lazy-plugins plugin enabled I lose all my custom key bindings.

### ISSUE THREAD

**Issue by doublethefish** [alangrainger/obsidian-lazy-plugins#6]
## The issue
With the lazy-plugins plugin enabled I lose all my custom key bindings.

The HotKey config is ignored by Obsidian, and HotKeys are set to defaults - in the UI, not the config file. This only happens with lazy-plugins enabled. 

For example: my custom key binding for `⌘B` (on OSX)  reverts back to the default to `editor:toggle-bold`, whilst my configuration file (on disk) remains the way I have configured it.

It is clear that the Hotkeys config file is just not being used by Obsidian when lazy-plugins is enabled.

**With the plugin enabled**
- My configuration file does NOT change, it remains configured as `"editor:toggle-bold": [],`
- The binding in the settings' `Hotkeys` panel is labeled as `Toggle bold`.

**With the plugin disabled**
- My Hotkey configuration works again.

## Expected behaviour
Hotkeys remain the same with the plugin enabled.

## Additional
I have not yet tried changing the Hotkey config in the UI, to see if that the overwrites my custom config with defaults.

**Comment by alangrainger** [comment]
Can you manually install [version 1.0.10](https://github.com/alangrainger/obsidian-lazy-plugins/releases/tag/1.0.10) (just copy the `main.js` and `manifest.json` files into your `.obsidian/plugins/lazy-plugins/` folder).

I'm wondering if the change [suggested by Obsidian team](https://github.com/obsidianmd/obsidian-releases/pull/3998#issuecomment-2318012016) actually does stop some plugins from working correctly.

1. Install v1.0.10
2. Set Hotkey plugin to be **Instant** load type
3. Restart Obsidian 2 times just to make sure everything is set correctly.
4. Test the hotkeys

**Comment by alangrainger** [comment]
Are you talking about the built-in Obsidian hotkeys functionality, or are you talking about a community plugin?

I can confirm that I do not lose my hotkeys configuration using this plugin, and I can't see any way at all that this plugin could affect your hotkeys.

Let me know, thanks

**Comment by doublethefish** [comment]
> Are you talking about the built-in Obsidian hotkeys functionality, or are you talking about a community plugin?
Yep. I am talking about the in-built Obsidian hotkeys.

![image](https://github.com/user-attachments/assets/d682166c-7532-4ec8-bfba-22be84ce5546)

> Can you manually install...
I will try to do this soon-ish.

**Comment by doublethefish** [comment]
Welp, to add to the confusion it actually appears to be non-deterministic, for example I can not repro with `⌘B` (toggle-sidebar) 100%  of the time , and my other mappings like `⌥⌘J` (for a plugin), or `⌥⬆` (move-line-up) disappear. In fact it seems non-deterministic which key mappings disappear - oh, actually it also looks like some of the slots for plug-in hotkeys completely disappear too, not always though. BUT the config file itself stays unmolested.

NOTE: I am noticing the disappearing binding/plug-in hotkeys when restarting Obsidian; which I was obviously doing a lot of to test start-time improvements.

I have also tried with a single key-bindings and just lazy installed in a clean Vault and it doesn't happen (as often??). The Vault where the problem exists has ~1000 notes, quite a few plug-ins installed and many re-bindings of keys. And vim-mode enabled with the obsidian-vimrc which modifies keybindings too.

I am currently on Sonoma 14.6.1, M1. Obsidian 1.6.7.

**Comment by alangrainger** [comment]
I recommend you have a look through the Lazy Loader code; all it does is call the built-in Obsidian functions `enablePlugin` and `disablePlugin`. I find it hard to imagine how that could affect hotkey mappings, and I certainly don't experience that on my own vault of 5000 notes and 40+ plugins.

I'm not saying it doesn't cause a conflict - it indeed may be causing this somehow - I just can't imagine _how_ it could be doing that.

If you manage to track down the cause please let me know :)

**Comment by doublethefish** [comment]
No worries, totally understand.

As there’s very little value for me here I probably won’t deep-dive.

**Comment by doublethefish** [comment]
This issue looks like it was an issue with the `toggle-list` plugin that was spotted by Obsidian and they disabled that plugin; see: https://github.com/thingnotok/obsidian-toggle-list/issues/36 I have also submitted a PR for that plugin and I have tested my repo case (above) and lazy now seems to work without issue. I will now have a look to see if it saves any load time for me.

This plug-in would have exacerbated that bug, because it changes the plugin load order.

It did seem like it was effecting more than just CMDs, but I've not looked in to that.
